### DISCUSSION TITLE
Dll import fails with IronPython 3.4

### DISCUSSION THREAD

**Discussion by rajesh78** [IronLanguages/ironpython3#1744] [Q&A]
Hi,
i have a created a dll in .Net using VS22
namespace NewTest
{
    public class Class1
    {
        public int Get_sum(int x, int y)
        {
            return x + y;
        }
    }
} 

i have installed ironPython 3.4
When i tried importing the dll using clr.AddReference it is giving error.
python code ----->
import clr
import os
import sys
import timeit
sys.path.append(os.getcwd())
#print(sys.path)
clr.AddReference("NewTest.dll")

i am getting error as below

Traceback (most recent call last): File "callcsharpdll.py", line 7, in FileNotFoundError: [Errno 2] Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

Please suggest

**Comment by BCSharp** [comment]
_.NET_ and _.NET Framework_ are binary incompatible with each other. You cannot use them both in the same process. Choose one or the other for your project and stick with the choice.

1. If you choose _.NET Framework_, set the target framework in your `NewTest.csproj` to the moniker starting with `net4` (e.g. `net481`), the version being equal or greater to the framework used by your `ipy` installation (to find out that, run `ipy -VV`, I bet in your case it will report `.NETFramework,Version=v4.6.2`). You also need to have installed the tooling targeting the framework version you will use in your project.
  Or, better yet, for this small example code, start a new project from scratch and select as project template "Class Library (.NET Framework)". This will create a `.csproj` file in the old MSbuild format and select the target framework version for which you already have tooling installed.

3. If you choose _.NET_, install IronPython for the latest version of .NET, and use that `ipy` to run your test. The link to instructions on installation for various .NET versions is on the home page of this GitHub project.

**Comment by rajesh78** [comment]
Thanks for your reply.
Point No 1 that you have given above is working for me.

My concern was for Point No 2. Where i was getting the error that I posted above.

I created a fresh project and i am getting this error. 

I have donwloaded the latest iron python msi and installed in my laptop. and i am getting this  error
Traceback (most recent call last): File "callcsharpdll.py", line 7, in FileNotFoundError: [Errno 2] Could not load file or assembly 'System.Runtime, Version=6.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified.

Request some help on this

**Reply by BCSharp** [reply]
You haven't followed the instructions in Point 2 as described. If you installed IronPython from an .msi, you installed ipy for _.NET Framework_, not for _.NET_. Follow the instructions for _.NET (Core)_ from the installation document. The easiest way is to install it as a dotnet tool.

**Reply by rajesh78** [reply]
Thanks I am able to call the dotnet dll now. I used the zip file to download and followed the installation instructions. My above code is now able to run.

But I am facing a problem in calling an api which has mssql database query, using entity frame work.
The error is showing like -- SystemError: Microsoft.Data.SqlClient is not supported on this platform.

would you please help on this
